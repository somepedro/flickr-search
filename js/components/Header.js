import React from 'react'
import { string, func } from 'prop-types'
import fetchJsonp from 'fetch-jsonp'
import { debounce, buildSearchUrl } from '../helpers'

class Header extends React.Component {
  constructor (props) {
    super(props)
    this.onTyping = this.onTyping.bind(this)
    this.onTypingCompleted = debounce(this.onTypingCompleted, 500)
    this.search = this.search.bind(this)
    this.state = {
      query: ''
    }
  }

  componentDidMount () {
    this.search(this.props.initialQuery)
  }

  onTyping (event) {
    const query = event.target.value

    if (query) {
      this.props.onQuerySent()
    }

    this.setState({ query })
    this.onTypingCompleted(query)
  }

  onTypingCompleted (searchQuery) {
    this.search(searchQuery)
  }

  search (searchQuery) {
    this.setState({ searchQuery })
    if (!searchQuery) {
      this.props.onResultsReceived(searchQuery, [])
    } else {
      try {
        const searchUrl = buildSearchUrl(searchQuery)
        fetchJsonp(searchUrl, {
          jsonpCallbackFunction: 'jsonFlickrApi'
        }).then((searchResults) => {
          return searchResults.json()
            .then((searchResultsJSON) => {
              this.props.onResultsReceived(searchQuery, searchResultsJSON.photos.photo)
            }).catch((error) => {
              console.log(error)
            })
        }).catch((error) => {
          console.log(error)
        })
      } catch (error) {
        console.log(error)
      }
    }
  }

  render () {
    return (
      <div className='header'>
        <input className='search-box' onChange={this.onTyping} value={this.state.query} type='text' placeholder='Search ...' />
      </div>
    )
  }
}

Header.propTypes = {
  initialQuery: string,
  onQuerySent: func,
  onResultsReceived: func
}

export default Header
