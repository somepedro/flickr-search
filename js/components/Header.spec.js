/* global test, expect */
import React from 'react'
import { shallow } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'
import Header from './FlickrSearch'

test('Header snapshot test', () => {
  const component = shallow(<Header />)
  const tree = shallowToJson(component)
  expect(tree).toMatchSnapshot()
})
