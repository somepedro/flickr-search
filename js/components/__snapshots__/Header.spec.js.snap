// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header snapshot test 1`] = `
<div
  className="flickr-search"
>
  <div
    className="stage"
  >
    <Header
      onQuerySent={[Function]}
      onResultsReceived={[Function]}
    />
    <Header
      onNavigateLeft={[Function]}
      onNavigateRight={[Function]}
      photo={null}
    />
  </div>
  <Slideshow
    chosenPhotoIndex={0}
    className="thumbnails"
    onPhotoClick={[Function]}
    photos={Array []}
  />
</div>
`;
